regparser/builder.py

```diff
@@ -84,16 +84,25 @@ def build_notices(self, checkpoint=True):
         for notice_group in self.eff_notices.values():
             self.notices.extend(notice_group)
 
-    def write_notices(self):
-        for notice in self.notices:
-            #  No need to carry this around
-            del notice['meta']
-            self.writer.notice(notice['document_number']).write(notice)
-
-    def write_regulation(self, reg_tree):
-        self.writer.regulation(self.cfr_part, self.doc_number).write(reg_tree)
-
-    def gen_and_write_layers(self, reg_tree, act_info, cache, notices=None):
+    def write_notices(self, output_type='json'):
+        if output_type == 'json':
+            for notice in self.notices:
+                #  No need to carry this around
+                del notice['meta']
+                self.writer.notice(notice['document_number']).write(notice)
+        elif output_type == 'xml':
+            for notice in self.notices:
+                del notice['meta']
+                self.writer.notice(notice['document_number']).write_notice(notice)
```

Comment:
Again: multiple levels of indirection make this a pain in the ass to resolve. Probably, yes, we could do it that way. Then we'll have to spend some time thinking about how to restructure that whole inheritance chain and calls to it so that it makes sense. I don't think it's worth the bother for the sake of saving an if-else.